processor/add_orchestrator_metadata/cluster_metadata_provider.go

```diff
@@ -15,34 +15,48 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package add_cluster_id
+package add_orchestrator_metadata
 
 import (
 	"context"
+	"github.com/elastic/elastic-agent-autodiscover/kubernetes/metadata"
+	agentconfig "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	k8s "k8s.io/client-go/kubernetes"
 )
 
 type ClusterHelper interface {
-	ClusterId() string
+	GetClusterMetadata() ClusterMetadata
 }
 
-type clusterHelper struct {
-	clusterId string
+type ClusterMetadataProvider struct {
+	metadata ClusterMetadata
+	logger   *logp.Logger
 }
 
-func newClusterHelper(client k8s.Interface) (ClusterHelper, error) {
+type ClusterMetadata struct {
+	clusterId   string
+	clusterName string
+}
+
+func newClusterMetadataProvider(client k8s.Interface, cfg *agentconfig.C, logger *logp.Logger) (ClusterHelper, error) {
```

Comment:
Also could that help if we have the option to set default values in the processor's config so as to use them as fallbacks in case we cannot retrieve the actual information from the provider/cluster? 

In Agent/Meatricbeat/Filebeat  we suggest to our users to do it by using the `add_fields` processor to manually define and add the fields in case those are not available. Here in this processor would it make sense to have the option to set the dafaults directly?